datasets/flwr_datasets/federated_dataset.py

```diff
@@ -102,6 +103,15 @@ def __init__(
         self._dataset: Optional[DatasetDict] = None
         # Indicate if the dataset is prepared for `load_partition` or `load_split`
         self._dataset_prepared: bool = False
+        event(
+            EventType.FEDERATED_DATASET_CREATED,
+            {
+                "dataset_name": self._dataset_name,
+                "partitioners_ids": [
+                    id(partitioner) for partitioner in partitioners.values()
+                ],
+            },
+        )
```

Comment:
This should move out of the constructor. Perhaps call it lazily in `load_partition` and `load_split`?